spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapPropertySourceLocator.java

```diff
@@ -88,7 +89,8 @@ private MapPropertySource getMapPropertySourceForSingleConfigMap(ConfigurableEnv
 	}
 
 	private void addPropertySourcesFromPaths(Environment environment, CompositePropertySource composite) {
-		properties.getPaths().stream().map(Paths::get).filter(p -> {
```

Comment:
and unique here. 

addendum: later in the code, we do `collect(Collectors.toList())`, where the result is a `List<Path>`. imo, it's not safe to do `Collectors.toSet()` there, because `map(Paths::get)`, specifically `Paths::get` does not guarantee any specific implementation of `Path`, which might not override hashCode/equals. In practice, it will return a `UnixPath` that _does_ override those methods, but I would stay on the safer side here.